schematic/store/synapse.py

```diff
@@ -680,105 +704,53 @@ def getFilesInStorageDataset(
         Raises:
             ValueError: Dataset ID not found.
         """
-        # select all files within a given storage dataset folder (top level folder in
-        # a Synapse storage project or folder marked with contentType = 'dataset')
-        walked_path = synapseutils.walk(
-            self.syn, datasetId, includeTypes=["folder", "file"]
-        )
+        file_list = []
 
-        current_entity_location = self.synapse_entity_tracker.get(
-            synapse_id=datasetId, syn=self.syn, download_file=False
-        )
+        # Get path to dataset folder by using childern to avoid cases where the dataset is the scope of the view
+        child_path = self.storageFileviewTable.loc[
```

Comment:
Should we account for cases where `self.storageFileviewTable` is empty? If self.storageFileviewTable is empty, then it would trigger a key error.

Reply:
In addition, I am curious can we assume that all file views have "path" column? I took a look at file view: syn52156826, and noticed that it doesn't have "path" column. If we can't assume that all file views have "path" column, can we add some logic here to better handle this situation?

Reply:
Thanks for the review, before going further, please review all comments.

1. I like the idea of handling empty fileview
2. There is error handling in the store/synapse.py for fileviews missing the path column (I think)

Reply:
@thomasyu888 Thanks for the feedback! Are you referring to this line: https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/store/synapse.py#L407? This line is only triggered if self.syn.tableQuery fails. However, even when the "path" column is missing, the query can still run successfully, so the exception here doesn’t catch that error. Apologies, I missed the JIRA comment [here](https://sagebionetworks.jira.com/browse/SCHEMATIC-183?focusedCommentId=224847). If we can ensure that all the file views we're currently using include the "path" column, and we don't need to consider legacy file views, then we shouldn't have to worry too much about the comment that I mentioned above.

Reply:
@linglp right, so you're asking if we can first validate if the fileview even has that column before running the query?  I think that could be a separate ticket if we end up tackling that.